torchdata/datapipes/iter/load/online.py

```diff
@@ -205,18 +220,33 @@ class OnlineReaderIterDataPipe(IterDataPipe[Tuple[str, StreamWrapper]]):
     """
     source_datapipe: IterDataPipe[str]
 
-    def __init__(self, source_datapipe: IterDataPipe[str], *, timeout: Optional[float] = None) -> None:
+    def __init__(
+        self,
+        source_datapipe: IterDataPipe[str],
+        *,
+        timeout: Optional[float] = None,
+        skip_on_error: bool = False,
+        **kwargs: Optional[Dict[str, Any]],
+    ) -> None:
         self.source_datapipe = source_datapipe
         self.timeout = timeout
+        self.skip_on_error = skip_on_error
+        self.query_params = kwargs
 
     def __iter__(self) -> Iterator[Tuple[str, StreamWrapper]]:
         for url in self.source_datapipe:
-            parts = urllib.parse.urlparse(url)
-
-            if re.match(r"(drive|docs)[.]google[.]com", parts.netloc):
-                yield _get_response_from_google_drive(url, timeout=self.timeout)
-            else:
-                yield _get_response_from_http(url, timeout=self.timeout)
+            try:
+                parts = urllib.parse.urlparse(url)
+
+                if re.match(r"(drive|docs)[.]google[.]com", parts.netloc):
+                    yield _get_response_from_google_drive(url, timeout=self.timeout, **self.query_params)
+                else:
+                    yield _get_response_from_http(url, timeout=self.timeout, **self.query_params)
```

Comment:
Can you please wrap `try-except` around `_get_response_from_google_drive` or `_get_response_from_http`?
In your current implementation, there is a chance the skipped Error comes from `parts = urllib.parse.urlparse(url)` or `re.match(r"(drive|docs)[.]google[.]com", parts.netloc)`.

Reply:
I will import and merge this after this change. Thanks!

Reply:
I have implemented this change, but I don't really understand why it is necessary. In my opinion the source of the exception doesn't really matter if we want to skip over them anyway. With this change exceptions caused by trying to parse the url will not be caught.

Reply:
> With this change exceptions caused by trying to parse the url will not be caught.

I think that is the point. If the URL cannot be parsed, perhaps users want to know and fix it. If you cannot get a response, then they may want to skip it.

Reply:
> With this change exceptions caused by trying to parse the url will not be caught.

I think that is the point. If the URL cannot be parsed, perhaps users want to know and fix it. If you cannot get a response, then they may want to skip it.